### PR TITLE
Added basic tests for osd metrics exporter

### PR DESCRIPTION
**Pull request by arjunrn**
Added basic tests for [osd-metrics-exporter](https://github.com/openshift/osd-metrics-exporter) based on existing tests.
Also a test to verify that the exporter is exposing metrics on an endpoint through a service.

### CODE REVIEW

**Comment by jharrington22** [comment]
@arjunrn Is this still good to merge?

**Comment by arjunrn** [comment]
@jharrington22 yes, it's still good.

**Review by jharrington22 — APPROVED** [review]

**Comment by jharrington22** [comment]
@jeefy @whereswaldon could you please take a look? Thanks!

File: pkg/e2e/operators/osd_metrics_exporter.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	)
+	h := helper.New()
+	checkClusterServiceVersion(h, operatorNamespace, operatorName)
+	checkDeployment(h, operatorNamespace, operatorName, 1)
+	checkClusterRoles(h, clusterRoles, true)
+	checkClusterRoleBindings(h, clusterRoleBindings, true)
+	checkUpgrade(helper.New(), operatorNamespace, operatorName, operatorName, "osd-metrics-exporter-registry")
+})
+
+var _ = ginkgo.FDescribe(osdMetricsExporterEndpointTest, func() {
```
Comment (whereswaldon) [inline pkg/e2e/operators/osd_metrics_exporter.go:46]:
```suggestion
var _ = ginkgo.Describe(osdMetricsExporterEndpointTest, func() {
```
Should this be focused? I can't think why, but I'm happy to be educated :D

**Review by whereswaldon — CHANGES_REQUESTED** [review]
One quick question on this before we hit the big green button :D

File: pkg/e2e/operators/osd_metrics_exporter.go (hunk 1 above)
Comment (arjunrn) [reply pkg/e2e/operators/osd_metrics_exporter.go:46]: It shouldn't be. It was left over after debugging.

**Comment by jharrington22** [comment]
Nice catch @whereswaldon. Thanks for fixing @arjunrn 

/lgtm

**Review by whereswaldon — COMMENTED** [review]
/lgtm

**Pull request merged**